src/components/TextField/TextField.tsx

```diff
@@ -0,0 +1,36 @@
+import React, { Component } from 'react'
+import Form from 'react-bootstrap/Form'
+import { Props } from './interfaces'
+
+/**
+ * A customizable text field component. It's a wrapper component built upon react's form controls.
+ */
+class TextField extends Component<Props, { textFieldValue: string }> {
```

Comment:
Currently, the convention that we are following is to have the Props interface defined in the same class the component is defined instead of defining it in the `interfaces.ts` file. This helps with some documentation generation things.

Reply:
This has been addressed.